src/delay.rs

```diff
@@ -70,30 +58,23 @@ impl Delay {
         // timer, meaning that we'll want to immediately return an error from
         // `poll`.
         if inner.list.push(&state).is_err() {
-            return Delay {
-                state: None,
-                when: at,
-            };
+            return Delay { state: None };
         }
 
         inner.waker.wake();
-        Delay {
-            state: Some(state),
-            when: at,
-        }
+        Delay { state: Some(state) }
     }
 
     /// Resets this timeout to an new timeout which will fire at the time
     /// specified by `at`.
     #[inline]
-    pub fn reset(&mut self, at: Instant) {
-        self.when = at;
-        if self._reset(self.when).is_err() {
+    pub fn reset(&mut self, delay: Duration) {
```

Comment:
Calling the param here `dur:` might be a clearer signifier of what is being passed:

```suggestion
    pub fn reset(&mut self, dur: Duration) {
```